libs/wingsdk/src/target-tf-azure/bucket.inflight.ts

```diff
@@ -83,6 +83,40 @@ export class BucketClient implements IBucketClient {
     return JSON.parse(await this.get(key));
   }
 
+  public async tryPut(key: string, body: string): Promise<boolean> {
+    try {
+      await this.put(key, body);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  public async tryPutJson(key: string, body: Json): Promise<boolean> {
+    try {
+      await this.putJson(key, body);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  public async tryGet(key: string): Promise<string | undefined> {
+    try {
+      return await this.get(key);
+    } catch (error) {
+      return undefined;
+    }
+  }
+
+  public async tryGetJson(key: string): Promise<Json | undefined> {
+    try {
+      return await this.getJson(key);
+    } catch (error) {
+      return undefined;
+    }
```

Comment:
Why do the targets behave differently? 
Are we checking only the case in which the key already exists? or any other case is counted too?

Reply:
They behave differently because the `cloud.Bucket` implementation for `tf-azure` target misses an `exist` method compared to `aws-shared` and `sim`.
Regarding your second question, I think we are just checking if a key already exists.

Reply:
They behave differently because the `cloud.Bucket` implementation for `tf-azure` target lacks an `exist` method compared to `aws-shared` and `sim`.
Regarding your second question, I think we are just checking if a key already exists.

Reply:
If so, let's add the `exist` method to azure too so they all behave the same :)
another option is to skip the implementation for azure for now, 

Another question that I have now is, if the `try` role is to check if the key exists, why do we need the `tryPut`? maybe can we omit it for now? what do you think?

Reply:
The Azure implementation is mandatory or a test would fail.
As for removing `tryPut`, I actually thought that was something you decided to add internally, since it was requested by @revitalbarletz on Slack. But if you changed your mind and that actually not needed, I'm fine with removing it.

Reply:
The Azure implementation is mandatory or a test would fail.

As for removing `tryPut`, I actually thought that was something you decided to add internally, since it was requested by @revitalbarletz on Slack. But if you changed your mind and that's actually not needed, I'm fine with removing it.

Reply:
it'll be great, thanks! We apologize for the confusion :)